jupyterhub/apihandlers/groups.py

```diff
@@ -37,7 +37,18 @@ class GroupListAPIHandler(_GroupAPIHandler):
     @admin_only
     def get(self):
         """List groups"""
-        data = [self.group_model(g) for g in self.db.query(orm.Group)]
+        offset = self.get_argument("offset", None)
+        limit = self.get_argument("limit", None)
+
+        query = self.db.query(orm.Group)
+
+        if offset is not None:
+            query = query.offset(int(offset))
```

Comment:
Need some argument validation to avoid 500 error on invalid offset, limit. Generally `raise web.HTTPError(400, "informative message")`